File: method1/src/cpp/engine_ext.cpp
#include "engine_ext.h"

// Global pointer to the engine
static QQmlApplicationEngine* g_engine = nullptr;

void set_global_engine(QQmlApplicationEngine* engine) {
	g_engine = engine;
}

void reload_qml_cache() {
	if (g_engine) {
		g_engine->clearComponentCache();
		qDebug() << "Refreshed QML cache!";
	}
}

File: method1/src/engine_ext.rs
#[cxx::bridge]
pub mod ffi {
	unsafe extern "C++" {
		include!("cxx-qt-lib/qqmlapplicationengine.h");
		type QQmlApplicationEngine = cxx_qt_lib::QQmlApplicationEngine;

		include!("engine_ext.h");

		unsafe fn set_global_engine(engine: *mut QQmlApplicationEngine);
		fn reload_qml_cache();
	}
}
